### ISSUE TITLE
Crash on Vulkan startup

### ISSUE THREAD

**Issue by satellitnorden** [baldurk/renderdoc#867]
Hi! I am trying out RenderDoc but cannot get it to work. I tried injecting into the process, before any API call to Vulkan is made, and then just continue from there. I fail to create my logical device though, and I see this in my log;

"RDOC 011488: [21:34:30]          vk_core.cpp( 955) - Warning - Spec versions of VK_KHR_dedicated_allocation are different between supported extension (1) and reported (3)!
RDOC 011488: [21:34:30]          vk_core.cpp( 955) - Warning - Spec versions of VK_KHR_swapchain are different between supported extension (68) and reported (69)!"

Which seems to be related. This suggests that my hardware does not support some layers that RenderDoc needs? I just want to verify if this is the case or if I can do anything to make it work. Thanks! I'll be happy to provide any source code you want, actually all of mine is publically available. Just ask. (:

**Comment by satellitnorden** [comment]
Please note that my application crashes during Capture Log as well. I can run my .exe fine outside of RenderDoc, but it crashes when launching through RenderDoc, possibly because of the above reason.

**Comment by baldurk** [comment]
Yes, please if you can share your application that would be ideal. Then I can investigate from there. Can you also clarify which branch you are running (v0.x or v1.x)?

Those warnings are almost certainly harmless - just indicating that the driver is compiled with a newer spec header than mine. The version numbers generally don't indicate breaking changes.

**Comment by satellitnorden** [comment]
I am currently using the 0.91 version of RenderDoc. (: This is my Github repository; https://github.com/satellitnorden/Catalyst-Engine You **should** be able to just clone it and run it, but, I haven't really tested it on more computers than two (my home computer and my work computer) so you might be running into some trouble. If so, let me know. (:

**Comment by baldurk** [comment]
Building straight out of the repository I ran into a problem with xinput1_4.dll being missing. Looking around it seems like that version isn't on windows 7 so I copied in xinput.lib from the June 2010 DX SDK and I was able to get it to link and run.

When I run it however it doesn't seem to present to the window, it just shows up as empty forever, even when I can verify by breakpointing that it is running through your code and isn't stuck:

![image](https://user-images.githubusercontent.com/661798/35943092-e1ffecf0-0c4f-11e8-8bc8-a16b2251017f.png)

Weirdly, if I then run with renderdoc it actually works *better*:

![image](https://user-images.githubusercontent.com/661798/35943206-36c1efb8-0c50-11e8-963d-6a902ce43a25.png)

(I'm not sure if that's how it's supposed to look or not).

FWIW I'm on win7, GTX 1080 with driver 388.59. I can test on other hardware tomorrow if that would help.

Where exactly do you get the problem, you say the VkDevice fails to create - do you mean it returns an error code?

**Comment by satellitnorden** [comment]
Ah, yes, the xinput.dll is part of the Windows 10 SDK, which explains you not finding it. Good that you brought that up, I should probably explicitly put that in the resository, thanks.
I have no idea why it does not render when you eventually run, though. I guess the Validation Layers say nothing? Those are disabled in the "Final" configuaration, so if you built with that, try "Release" or "Debug" instead. 
And yes, what you are seeing there is my mildly misfigured terrain. It's weird that the skybox renders correctly for you though, it does not for me, which I was originally wanting to debug with RenderDoc. (: I am running with an AMD Radeon R9 280X graphics card, btw, latest drivers. Don't know it that helps you.

When injecting into the process with RenderDoc, I get an access violation on the "vkCreateDevice()". I get this error message right before; "vkCreateDevice: pCreateInfo->pQueueCreateInfos[0].queueCount (=3) is not less than or equal to available queue count for this pCreateInfo->pQueueCreateInfos[0].queueFamilyIndex} (=0) obtained previously from vkGetPhysicalDeviceQueueFamilyProperties (i.e. is not less than or equal to 1). The spec valid usage text states 'queueCount must be less than or equal to the queueCount member of the VkQueueFamilyProperties structure, as returned by vkGetPhysicalDeviceQueueFamilyProperties in the pQueueFamilyProperties[queueFamilyIndex]' (https://www.khronos.org/registry/vulkan/specs/1.0/html/vkspec.html#VUID-VkDeviceQueueCreateInfo-queueCount-00382)"

On further investigation, it seems to fault already when picking queue family indices, it seems that when I inject into the process, "vkGetPhysicalDeviceQueueFamilyProperties()" reports only 1 queueFamilyCount, wheres running the app standalone it reports 3.. I'm going to see if I can backtrack this further so that I could provide you with more info.

**Comment by baldurk** [comment]
The link library for xinput was there and I was able to link the exe, but then when running I didn't have xinput1_4.dll. As far as I can tell from googling, this version of xinput is just not available on windows 7 - I have the windows 10 SDK but it doesn't include that dll.

I didn't get any debug output from the validation layers, but I did build the Debug configuration. I saw them load up but when I looked I couldn't find where you create a debug report object, am I missing something?

The queue family thing is expected. RenderDoc doesn't support multiple queues, so it changes the reported families and queue count to only expose a single queue. Your code will need to adapt to that, since it's possible some drivers will only expose one queue also.

**Comment by satellitnorden** [comment]
Ah, yes, I see. That explains. I whipped some temporary one-queue support to test it out, and it seems that now RenderDoc injects successfully into the process and is able to capture a frame! I can start debugging now. Thanks a ton. I'm kind of embarassed it's such a simple thing as not supporting one queue, but it's working now. So thanks a ton! I'll close this now. (:

**Comment by satellitnorden** [comment]
As a sidenote, using RenderDoc, I managed to debug why I experienced the problems I had. This was my initial state;

![yes](https://user-images.githubusercontent.com/7903189/35947380-f4dfb4be-0c67-11e8-9568-73e8a8b61c87.jpg)

And this is the current state;

![yas](https://user-images.githubusercontent.com/7903189/35947454-3e556224-0c68-11e8-9c3c-c08222b03809.jpg)

The terrain mesh is still messed up beyond words, but my SkyBox renders as it should now! Thank you for making such a great tool. I will definitely be using this extensively to debug in the future. (:
